src/main.ts

```diff
@@ -77,10 +80,10 @@ export default class TTSPlugin extends Plugin {
 		this.registerEvent(this.app.workspace.on('editor-menu', ((menu, editor, markdownView) => {
 			menu.addItem((item) => {
 				item
-					.setTitle("Say selected text")
+					.setTitle(window.getSelection().toString().length > 0 ? "Read selected text" : "Read the note")
 					.setIcon("audio-file")
 					.onClick(() => {
-						this.ttsService.say("", editor.getSelection(), this.ttsService.getLanguageFromFrontmatter(markdownView));
+						this.ttsService.play(markdownView);
```

Comment:
ttsService.play() method will do the necessary checks to read the selected texts. No need to use different methods anymore.